fundamental-concepts/choices-role-pattern/3_consuming_choices.md

```diff
@@ -0,0 +1,113 @@
+  }}' localhost:7575/v1/create
+
+```{{execute T1}}
+
+and get a response
+
+```
+{"result":{"agreementText":"","contractId":"0060809eb719f9d28a2d783369440d85630dbc8069c8d7b859556cf8765e4a0858","observers":["Bob"],"payload":{"offeringParty":"Alice","receivingParty":"Bob","offer":"Get
+Alices Barbeque Sauce 50%
+off!"},"signatories":["Alice"],"templateId":"78b4e7ce7dc580ccbc77aa2d05b6659e2df24047af5f72db131cfb0b56d6a943:User:SpecialOffer"},"status":200}%
+```
+
+Once the receiving party takes your offer, you'd like the offer to be removed from the ledger.
+Otherwise `Bob` would get `Alice` barbeque sauce for half the price forever! A good way to get this
+semantic would be to add a `consuming` choice `TakeOffer` to the `SpecialOffer` template:
+
+<pre class="file" data-target="clipboard">
+    choice TakeOffer: ()
+      controller receivingParty
+        do
+          return ()
+</pre>
+
+Note that we dropped the `nonconsuming` before the `choice` keyword. This makes the choice
+`consuming`.
+
+Because the `receivingParty` is the controller of the choice only she can take the offer (check out
+the previous [Katacoda](https://daml.com/learn/fundamental-concepts/template-authorization) for more
+on authorization and controllers). As soon as she takes the offer, the `SpecialOffer` is archived on
+the ledger, marked as inactive and the choice can not be executed any longer.
+
+```
+curl -X POST -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJCb2IiXX19.zU-iMSFG90na8IHacrS25xho3u6AKnSlTKbvpkaSyYw' -d '{
+    "templateId": "User:SpecialOffer",
+    "contractId": "<CONTRACT_ID>",
+    "choice": "TakeOffer",
+    "argument": {
+    }
+}' localhost:7575/v1/exercise
+
+```{{execute T1}}
+
+Again, you have to replace the `CONTRACT_ID` placeholder with the previously returned contract ID of
+the `SpecialOffer` contract. The sandbox responds that it has indeed archived the `SpecialOffer`
+contract and that the return value of the choice is empty, exactly as specified in the `TakeOffer`
+choice:
+
+```
+{"result":{"events":[{"archived":{"contractId":"0060809eb719f9d28a2d783369440d85630dbc8069c8d7b859556cf8765e4a0858","templateId":"78b4e7ce7dc580ccbc77aa2d05b6659e2df24047af5f72db131cfb0b56d6a943:User:SpecialOffer"}}],"exerciseResult":{}},"status":200}%
+```
+
+Now let's try to get more cheap barbeque sauce:
+
+```
+curl -X POST -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJCb2IiXX19.zU-iMSFG90na8IHacrS25xho3u6AKnSlTKbvpkaSyYw' -d '{
+    "templateId": "User:SpecialOffer",
+    "contractId": "<CONTRACT_ID>",
+    "choice": "TakeOffer",
+    "argument": {
+    }
+}' localhost:7575/v1/exercise
+
+```{{execute T1}}
```

Comment:
```suggestion
<pre class="file" data-target="clipboard">
curl -X POST -H "Content-Type: application/json" -H 'Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJodHRwczovL2RhbWwuY29tL2xlZGdlci1hcGkiOnsibGVkZ2VySWQiOiJNeUxlZGdlciIsImFwcGxpY2F0aW9uSWQiOiJmb29iYXIiLCJhY3RBcyI6WyJCb2IiXX19.zU-iMSFG90na8IHacrS25xho3u6AKnSlTKbvpkaSyYw' -d '{
    "templateId": "User:SpecialOffer",
    "contractId": "<CONTRACT_ID>",
    "choice": "TakeOffer",
    "argument": {
    }
}' localhost:7575/v1/exercise

</pre>
```

Same comment as before: users need to change the contractID before running it.

Reply:
will be done automatically.